### ISSUE TITLE
Do you have the evaluation code?

### ISSUE THREAD

**Issue by XUHAN314** [bengioe/gflownet#7]
I didn't find the evaluation or test code in the repo. How do you test the generative model?

**Comment by bengioe** [comment]
Hi, experiments typically produce a file called `sampled_mols.pkl.gz`:

https://github.com/GFNOrg/gflownet/blob/831a6989d1abd5c05123ec84654fb08629d9bc38/mols/gflownet.py#L384-L400

What's reported in the paper is the rolling top-k, which we compute after the fact by loading the molecules and computing the top-k-so-far during learning.

The results are stored as a list of 4-tuples:
https://github.com/GFNOrg/gflownet/blob/831a6989d1abd5c05123ec84654fb08629d9bc38/mols/gflownet.py#L210
So to get the ground reward you can take the `r` value and inverse-exponentiate it back to the reward prediction of the proxy `r ** (1 / args.reward_exp)`.

It would be nice to add a notebook with that kind of code. It shouldn't be too hard for me to take existing notebooks and clean them up, but I don't have time right now. I'll mark this issue as enhancement.

**Comment by XUHAN314** [comment]
Thanks for your reply. There are some plots in the paper, I am wondering how do you make them. It would be nice if you could create a notebook to show that. Thanks! This is a great work!

**Comment by XUHAN314** [comment]
Does Top K mean you selected K mols with the highest rewards? How do you select K?

**Comment by bengioe** [comment]
By Top K we mean something like `topk = lambda x, k: x[np.argsort(x)[-k:]]`. In the paper we report for `k` 10, 100, and 1000, where `x` is the set of rewards sampled so far.

**Comment by XUHAN314** [comment]
Thanks! It helps a lot.

**Comment by 58191554** [comment]
which file should I run first, in other words, where is the beginning?

**Comment by bengioe** [comment]
Simply running `python gflownet.py` in the `mols` directory should train a model with the default hyperparameters.
